src/app/serverViewState.ts

```diff
@@ -238,11 +238,11 @@ export class ServerViewState {
 
         let httpUrl = url;
         if (!isValidURL(url)) {
-            // If it already includes the protocol, tell them it's invalid
-            if (isValidURI(url)) {
+            // If it already includes the protocol, force it to HTTPS
+            if (isValidURI(url) && !url.startsWith('http')) {
                 httpUrl = url.replace(/^((.+):\/\/)?/, 'https://');
-            } else {
-                // Otherwise add HTTPS for them
+            } else if (!url.match(/^(h|ht|htt|http|https):?\/?\/?$/)) {
```

Comment:
I'd say this is less code though, because then i'd have to check all of the above cases for substrings and it would be a mess of ORs. I usually try to avoid regexes as well these days, but I think for this it makes sense.

Reply:
Maybe we could flip the check around? Something like `'https://'.startsWith(url)` seems like it would work, although we'd have to check both HTTP and HTTPS separately

Reply:
I meant a generic function, not a mess of ORs ;) But looks like Harrison's suggestion would work as well. Checking for both http and https doesn't sound like too much overhead.

Reply:
Harrison's solution sounds very clean so yes I will implement that instead.